### PR TITLE
Navigate Back to Intro Screen when Returning from App Permission Settings

### PR DESCRIPTION
**Pull request by iaulakh**
## Purpose

* Removing task affinity
* Restart IntroActivity when app was minimized and returning with changed permissions

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/JoinCallActivity.java [hunk 1]
```diff
@@ -25,6 +28,7 @@
 public class JoinCallActivity extends AppCompatActivity {
     private static final String LOG_TAG = JoinCallActivity.class.getSimpleName();
     private EditText editTextTextMeetingName;
+    private ActivityResultLauncher<Intent> mStartForResult;
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/JoinCallActivity.java:31]: If we handle everything in SetupActivity this should not be needed.

**Review by waynemo — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java [hunk 2]
```diff
@@ -178,8 +178,9 @@ private void openSettings() {
         intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         final Uri uri = Uri.fromParts("package", getPackageName(), null);
         intent.setData(uri);
-        finishAffinity();
         startActivity(intent);
+        setResult(RESULT_OK);
+        finish();
```
[the remark this answers is not in the document]
Comment (iaulakh) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java]: done

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/JoinCallActivity.java (hunk 1 above)
Comment (iaulakh) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/JoinCallActivity.java]: ok Wayne

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/EndActivity.java [hunk 3]
```diff
@@ -37,6 +38,28 @@ protected void onCreate(final Bundle savedInstanceState) {
...
 
         endTextView = findViewById(R.id.end_return);
-        endTextView.setOnClickListener(l -> finish());
+        endTextView.setOnClickListener(l -> {
+            startIntroActivity();
+            finish();
+        });
+    }
+
+    private void startIntroActivity() {
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/EndActivity.java:47]: We should include private methods after public and protected methods.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java [hunk 4]
```diff
@@ -164,21 +166,46 @@ protected void onDestroy() {
         super.onDestroy();
     }
 
+    protected void onResume() {
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java:169]: For new methods in this PR, let's make sure we order them
1. public
2. protected
3. private

For existing methods, we can create a new PR to reorder.

**Review by waynemo — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java (hunk 4 above)
Comment (iaulakh) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java]: ok

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/EndActivity.java (hunk 3 above)
Comment (iaulakh) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/EndActivity.java:59]: ok

**Review by waynemo — APPROVED** [review]

**Review by lpeixin — APPROVED** [review]

**Pull request merged**